.github/workflows/actions/module-compatibility-handler/index.js

```diff
@@ -0,0 +1,306 @@
+      // If the renamed module is the one we are checking, note its real name
+      if (renamedLinuxModules.get(linuxModules[i]) === module) {
+        realLinuxModuleName = linuxModules[i];
+      }
+      linuxModules[i] = renamedLinuxModules.get(linuxModules[i]);
+    }
+  }
+
+  // No rename occurred
+  if (realLinuxModuleName === '') {
+    realLinuxModuleName = module;
+  }
+
+  const compatibleLinuxModules = filterIncompatibleModules(editorVersion, linuxModules, incompatibleLinuxModules);
+
+  //Add on base to the requested modules and put into a set to ensure no duplicates
+  const moduleSet = new Set([...compatibleLinuxModules, 'base']);
+
+  // Return whether the module is supported and what the real name is for the module
+  return [moduleSet.has(module), realLinuxModuleName];
+}
+
+const isSupportedMacModule = async (editorVersion, editorChangeset, module)=> {
+  const macModules = await getModules(editorVersion, editorChangeset, 'osx');
+
+  let realMacModuleName = '';
+
+  // Fix any renamed modules
+  for (let i = 0; i < macModules.length; ++i) {
+    if (renamedMacModules.has(macModules[i])) {
+      // If the renamed module is the one we are checking, note its real name
+      if (renamedMacModules.get(macModules[i]) === module) {
+        realMacModuleName = macModules[i];
+      }
+      macModules[i] = renamedMacModules.get(macModules[i]);
+    }
+  }
+
+  // No rename occurred
+  if (realMacModuleName === '') {
+    realMacModuleName = module;
+  }
+
+  const compatibleMacModules = filterIncompatibleModules(editorVersion, macModules, incompatibleMacModules);
+
+  //Add on base to the requested modules and put into a set to ensure no duplicates
+  const moduleSet = new Set([...compatibleMacModules, 'base']);
+
+  // Return whether the module is supported and what the real name is for the module
+  return [moduleSet.has(module), realMacModuleName];
+}
+
+//Fetches all supported modules for a specific verion of unity on a given base platform
+async function getModules(version, changeset, platform) {
+  // Get the config file that details all modules available to the version
+  const res = await fetch(`${UNITY_DOWNLOAD_API_URL}/${changeset}/unity-${version}-${platform}.ini`);
+
+  if (res.status >= 400) {
+      throw new Error("Bad response from Unity API");
+  }
+
+  const responseBody = await res.text();
+  const moduleSet = new Array();
+
+  // Pull out each module name using regex
+  const reMatchedModules = responseBody.matchAll(reMatchUnityModuleNames);
+
+  // Add names to a set that we return
+  let module = reMatchedModules.next();
+  while(!module.done) {
+      moduleSet.push(module.value[1].toLowerCase());
+      module = reMatchedModules.next();
+  }
+  return moduleSet;
+}
+
+/**
+ * Compares Unity Versions
+ * Returns -1 if versionA is less than versionB
+ * Returns 1 if versionA is greater than versionB
+ * Returns 0 if the versions are identical
+ */
+function compareVersions(versionA, versionB) {
```

Comment:
I'm wondering if there's an easier way for version comparison. Also I feel like this could break for beta versions if we want to support that for the future (but hey I didn't test this at all)

Reply:
I just added support for looking at the version letter and the number after it. I'm assuming a comes before b which comes before f. Then finally compare by the number after the letter. Let me know if there is anything I am missing there as I am not too well versed with Unity's versioning scheme.